server/scala/serv/src/main/scala/com/ctc/omega_edit/grpc/EditorService.scala

```diff
@@ -33,41 +25,67 @@ import com.ctc.omega_edit.grpc.Session._
 import com.google.protobuf.ByteString
 import com.google.protobuf.empty.Empty
 import io.grpc.Status
-import omega_edit._
 
+import java.io.{BufferedInputStream, FileInputStream, IOException}
+import java.lang.management.ManagementFactory
 import java.nio.file.Paths
+import omega_edit._
+import org.apache.pekko
+import org.apache.tika.detect.DefaultDetector
+import org.apache.tika.metadata.Metadata
+import pekko.NotUsed
+import pekko.actor.ActorSystem
+import pekko.grpc.GrpcServiceException
+import pekko.http.scaladsl.Http
+import pekko.pattern.ask
+import pekko.stream.scaladsl.Source
+import pekko.util.Timeout
+
+import scala.concurrent.ExecutionContext
+import scala.concurrent.ExecutionContext.Implicits.global
 import scala.concurrent.duration.DurationInt
 import scala.concurrent.{Await, Future}
-
 import scala.util.{Failure, Success}
-import scala.concurrent.ExecutionContext.Implicits.global
-
-import scala.concurrent.ExecutionContext
-import java.lang.management.ManagementFactory
 
 class EditorService(implicit val system: ActorSystem) extends Editor {
   private implicit val timeout: Timeout = Timeout(5.seconds)
   private val editors = system.actorOf(Editors.props())
   private var isGracefulShutdown = false
   import system.dispatcher
 
+  private def detectFileType(filePath: String): Option[String] =
+    try {
+      val file = new BufferedInputStream(new FileInputStream(filePath))
+      val detector = new DefaultDetector()
+      val metadata = new Metadata()
+      val mediaType = detector.detect(file, metadata)
+      Option(mediaType.toString)
+    } catch {
+      case _: IOException => None
+    }
+
   def getVersion(in: Empty): Future[VersionResponse] = {
     val v = OmegaEdit.version()
     Future.successful(VersionResponse(v.major, v.minor, v.patch))
   }
 
   def createSession(in: CreateSessionRequest): Future[CreateSessionResponse] =
-    isGracefulShutdown match {
-      case false =>
-        (editors ? Create(
-          in.sessionIdDesired,
-          in.filePath.map(Paths.get(_))
-        )).mapTo[Result]
-          .map {
-            case Ok(id) => CreateSessionResponse(id)
-            case Err(c) => throw grpcFailure(c)
-          }
-      case true => Future.successful(CreateSessionResponse(""))
+    if (isGracefulShutdown) {
```

Comment:
Revert this back to the original and just update it

```scala
isGracefulShutdown match {
      case false =>
        (editors ? Create(
          in.sessionIdDesired,
          in.filePath.map(Paths.get(_))
        )).mapTo[Result]
          .map {
            case Ok(id) => 
              filepath match {
                case Some(path) => CreateSessionResponse(id, detectFileType(path.toString))
                case None => CreateSessionResponse(id, None)
            }
            case Err(c) => throw grpcFailure(c)
          }
      case true => Future.successful(CreateSessionResponse(""))
```

Its a lot cleaner of an update that way, the rewriting is unnecessary.

Reply:
IntelliJ actually linted this trivial match and refactored it to the simpler if/else in this PR.

<img width="761" alt="Screenshot 2023-04-17 at 15 12 20" src="https://user-images.githubusercontent.com/2205472/232587807-bbbb3e4d-9b1d-4c85-bf7e-f77922bf3ee3.png">

I say we go with the if/else.